invenio_pidstore/providers/datacite.py

```diff
@@ -31,20 +35,66 @@ class DataCiteProvider(BaseProvider):
     default_status = PIDStatus.NEW
     """Default status for newly created PIDs by this provider."""
 
+    doi_prefix_regexp = re.compile(
+        r"10\.\d+(\.\d+)*$"
+    )
+
     @classmethod
-    def create(cls, pid_value, **kwargs):
+    def valid_doi_prefix(cls, prefix):
+        """Matches if prefix is a DOI prefix.
+
+        NOTE: This was done here to prevent relying on idutils.
+        """
+        return cls.doi_prefix_regexp.match(prefix)
+
+    @classmethod
+    def generate_doi(cls, options=None):
+        """Generate DOI with random suffix."""
+        passed_options = options or {}
+        # WHY: A new dict needs to be created to prevent side-effects
+        options = dict(current_app.config.get(
+            'PIDSTORE_DATACITE_DOI_OPTIONS', {}
+        ))
+        options.update(passed_options)
+        prefix = options.get(
```

Comment:
There's some inconsistency here, that I think we have to fix. The init method creates an DataCite API client that only works with a preconfigured doi prefix (``PIDSTORE_DATACITE_DOI_PREFIX``), thus you can create a DOI with a prefix you can't register in. I would say that somehow the provider should be able to select the right credentials for a given prefix.

Reply:
From video chat, I will leave `DataCiteProvider` mostly as it was before these changes (except for added documentation) and I will create a separate provider that creates local `recid`. It will address that.